courses-app/src/components/Courses/Courses.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+
+import { BUTTONS_TEXT, mockedCoursesList } from '../../constants';
+
+import { Button } from '../../common/Button/Button';
+import CreateCourse from '../CreateCourse/CreateCourse';
+import CourseCard from './components/CourseCard/CourseCard';
+import SearchBar from './components/SearchBar/SearchBar';
+
+import { StyledCourses, StyledHeadingCourses } from './Courses.style';
+import { CourseType } from '../../types';
+
+const Courses = () => {
+	const [newCourse, setNewCourse] = useState(false);
+	const [courses, setCourses] = useState(mockedCoursesList);
+	const [filteredCourses, setFilteredCourses] = useState(courses);
+
+	const addingCourseHandler = () => {
+		setNewCourse((prev) => !prev);
+	};
+
+	const filteringCoursesHandler = (courses: CourseType[]) => {
+		setFilteredCourses(courses);
+	};
+
+	const creationCoursesHandler = (course: CourseType) => {
+		setCourses((prev) => [...prev, course]);
+		mockedCoursesList.push(course);
+		setNewCourse((prev) => !prev);
+	};
```

Comment:
usage of prev state here is redundant

`setNewCourse((prev) => !prev)` === `setNewCourse(!newCourse)`

Reply:
Change.